### PR TITLE
[rbmaradio] Fixed extractor

### PR DESCRIPTION
**Pull request by zvonicek**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This PR fixes extractor for rbmaradio.com. Due to recent changes it was not possible to download new episodes anymore. The PR fixes it and makes it functional again.

### CODE REVIEW

File: youtube_dl/extractor/rbmaradio.py [hunk 1]
```diff
@@ -31,25 +29,27 @@ def _real_extract(self, url):
...
-            'title': data['title'],
-            'description': data.get('teaser_text'),
-            'location': data.get('country_of_origin'),
-            'uploader': data.get('host', {}).get('name'),
-            'uploader_id': data.get('host', {}).get('slug'),
-            'thumbnail': data.get('image', {}).get('large_url_2x'),
-            'duration': data.get('duration'),
+            'title': item.get('title') + ' - ' + item.get('showTitle'),
+            'description': clean_html(item.get('longTeaser')),
+            'thumbnail': "http:" + item.get('imageURL', {}).get('landscape'),
```
Comment (dstftw) [inline youtube_dl/extractor/rbmaradio.py]: Use `self._proto_relative_url` instead.

File: youtube_dl/extractor/rbmaradio.py [hunk 2]
```diff
@@ -31,25 +29,27 @@ def _real_extract(self, url):
...
             raise ExtractorError('Invalid JSON: ' + str(e))
 
-        video_url = data['akamai_url'] + '&cbr=256'
+        item = None
+        for episode in data['episodes']:
+            items = data['episodes'][episode]
+            if video_id in items:
+                item = items[video_id]
+
+        video_url = item.get('audioURL') + '?cbr=256'
```
Comment (dstftw) [inline youtube_dl/extractor/rbmaradio.py]: Should be `item['audioURL']`.

**Comment by zvonicek** [comment]
Thanks for the review, @dstftw. I fixed the issues and squashed them into the previous commit.

File: youtube_dl/extractor/rbmaradio.py [hunk 3]
```diff
@@ -7,22 +7,20 @@
 from .common import InfoExtractor
 from ..utils import (
     ExtractorError,
+    clean_html
 )
 
 
 class RBMARadioIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?rbmaradio\.com/shows/(?P<videoID>[^/]+)$'
+    _VALID_URL = r'https?://(?:www\.)?rbmaradio\.com/shows/.*/episodes/(?P<videoID>[^/]+)$'
```
Comment (dstftw) [inline youtube_dl/extractor/rbmaradio.py]: This allows URLs like https://www.rbmaradio.com/shows//episodes/ford-lopatin-live-at-primavera-sound-2011` that is a priori invalid.

File: youtube_dl/extractor/rbmaradio.py [hunk 4]
```diff
@@ -31,25 +29,27 @@ def _real_extract(self, url):
         video_id = m.group('videoID')
 
         webpage = self._download_webpage(url, video_id)
-
-        json_data = self._search_regex(r'window\.gon.*?gon\.show=(.+?);$',
-                                       webpage, 'json data', flags=re.MULTILINE)
+        json_data = self._search_regex(r'<script>window.__INITIAL_STATE__ = (.+?)</script>',
```
Comment (dstftw) [inline youtube_dl/extractor/rbmaradio.py]: This should be relaxed to `window\.__INITIAL_STATE__\s*=\s*`.

File: youtube_dl/extractor/rbmaradio.py [hunk 5]
```diff
@@ -31,25 +29,27 @@ def _real_extract(self, url):
...
+                                       webpage, 'json data')
 
         try:
             data = json.loads(json_data)
```
Comment (dstftw) [inline youtube_dl/extractor/rbmaradio.py]: Use `self._parse_json` instead.

**Comment by zvonicek** [comment]
Ok, fixed.

**Pull request closed without merging**